### PR TITLE
add github action for publishing new releases

### PR DESCRIPTION
**Pull request by BayooG**

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# This workflows will upload a Python Package using Twine a new when a release is created
+# For more information see: https://help.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions#publishing-to-package-registries
+
+name: Publish on PYPI
+
+on:
+  release:
+    types: [created]
```
Comment (GRBurst) [inline .github/workflows/publish.yml:8]: let's run it on every time (every commit on main and every pr)

Comment (cornerman) [reply .github/workflows/publish.yml]: For the publish, we need a version tag, so we have a proper version for pip.
I would propose using tags, releases are not so nice to manage in the end.

**Review by cornerman — COMMENTED** [review]
I think, we should move this workflow to the other workflow file, so that we only publish when the tests actually run.

File: .github/workflows/publish.yml (hunk 1 above)
Comment (BayooG) [reply .github/workflows/publish.yml]: okay makes sense.
I'll switch to running this flow on new tags, and make sure it doesn't run unless the build flow succeeded.

**Pull request merged**